packages/analytics/lib/index.d.ts

```diff
@@ -50,143 +50,124 @@ import { ReactNativeFirebase } from '@react-native-firebase/app';
+    item_variant?: string;
   }
-
-  export interface AddToWishlistEventParameters {
+  export interface AddPaymentInfoEventParameters {
+    items?: Item[];
     /**
-     * An item ID.
+     * Purchase currency in 3 letter [ISO_4217](https://en.wikipedia.org/wiki/ISO_4217#Active_codes) format. E.g. `USD`.
      */
-    item_id: string;
+    currency?: string;
+    value?: number;
     /**
-     * An item name.
+     * Coupon code for a purchasable item.
      */
-    item_name: string;
+    coupon?: string;
     /**
-     * An item category.
+     * The chosen method of payment
      */
-    item_category: string;
+    payment_type?: string;
+  }
+
+  export interface AddShippingInfoEventParameters {
+    items?: Item[];
     /**
-     * Purchase quantity.
+     * Purchase currency in 3 letter [ISO_4217](https://en.wikipedia.org/wiki/ISO_4217#Active_codes) format. E.g. `USD`.
      */
-    quantity: number;
+    currency?: string;
+    value?: number;
     /**
-     * Purchase price.
+     * Coupon code for a purchasable item.
      */
-    price?: number;
+    coupon?: string;
     /**
-     * A context-specific numeric value which is accumulated automatically for each event type. Values
-     * can include revenue, distance, time and points. When a value is set, the accompanying `currency`
-     * parameter should also be defined.
+     * The shipping tier (e.g. Ground, Air, Next-day) selected for delivery of the purchased item
      */
-    value?: number;
+    shipping_tier?: string;
+  }
+
+  export interface AddToCartEventParameters {
+    items?: Item[];
     /**
      * Purchase currency in 3 letter [ISO_4217](https://en.wikipedia.org/wiki/ISO_4217#Active_codes) format. E.g. `USD`.
      */
     currency?: string;
     /**
-     * The Google [Place ID](https://developers.google.com/places/place-id) that corresponds to the associated item (String). Alternatively, you can supply your own custom Location ID.
+     * value of item
      */
-    item_location_id?: string;
+    value?: number;
   }
 
-  export interface BeginCheckoutEventParameters {
+  export interface AddToWishlistEventParameters {
+    items?: Item[];
     /**
-     * A context-specific numeric value which is accumulated automatically for each event type. Values
-     * can include revenue, distance, time and points. When a value is set, the accompanying `currency`
-     * parameter should also be defined.
+     * Purchase currency in 3 letter [ISO_4217](https://en.wikipedia.org/wiki/ISO_4217#Active_codes) format. E.g. `USD`.
      */
+    currency?: string;
     value?: number;
+  }
+
+  export interface BeginCheckoutEventParameters {
     /**
      * Purchase currency in 3 letter [ISO_4217](https://en.wikipedia.org/wiki/ISO_4217#Active_codes) format. E.g. `USD`.
      */
+    //TODO if value is a param, so must currency: https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.Event#public-static-final-string-add_to_wishlist
```

Comment:
TODO here

Reply:
I'm not actually sure about this. If `currency` is a property on the object then `value` must also be present.

I think we will have to update the event types to have something like this: https://stackoverflow.com/a/51507473 what do you think?

The `superstruct` object check should probably ensure if one property is present, the other one should be as well I think. Let me know if you think otherwise.